website/bot.py

```diff
@@ -0,0 +1,118 @@
+from pathlib import Path
+
+from dotenv import find_dotenv, load_dotenv
+from langchain.chains import ConversationalRetrievalChain
+from langchain.memory import ConversationSummaryMemory
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_community.document_loaders import (
+    DirectoryLoader,
+    Docx2txtLoader,
+    PyPDFLoader,
+    TextLoader,
+    UnstructuredMarkdownLoader,
+)
+from langchain_community.vectorstores import FAISS
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI, OpenAIEmbeddings
+
+load_dotenv(find_dotenv(), override=True)
+
+
+def load_document(file_path):
+    loaders = {
+        ".pdf": PyPDFLoader,
+        ".docx": Docx2txtLoader,
+        ".txt": TextLoader,
+        ".md": UnstructuredMarkdownLoader,
+    }
+
+    file_path = Path(file_path)
+    extension = file_path.suffix
+    Loader = loaders.get(extension)
+
+    if Loader is None:
+        raise ValueError(f"Unsupported file format: {extension}")
+
+    return Loader(file_path).load()
+
+
+def load_directory(dir_path):
+    return DirectoryLoader(dir_path).load()
+
+
+def split_document(chunk_size, chunk_overlap, document):
+    text_splitter = RecursiveCharacterTextSplitter(
+        chunk_size=chunk_size,
+        chunk_overlap=chunk_overlap,
+        length_function=len,
+    )
+    return text_splitter.split_documents(document)
+
+
+def embed_documents_and_save(embed_docs, db_dir="", db_name="faiss_index"):
+    db_path = Path(db_dir)
+    if not db_path.exists():
+        db_path.mkdir(parents=True, exist_ok=True)
+
+    db_file_path = db_path / db_name
+
+    try:
+        embeddings = OpenAIEmbeddings(model="text-embedding-3-small")
+        if db_file_path.exists():
+            db = FAISS.load_local(db_file_path, embeddings, allow_dangerous_deserialization=True)
+            db.add_documents(embed_docs)
+        else:
+            db = FAISS.from_documents(embed_docs, embeddings)
+
+        db.save_local(db_file_path)
+        return db
+    except FileNotFoundError as fnf_error:
```

Comment:
I wonder what part of code will throw `FileNotFoundError` here?
I see you check `db_file_path.exists()` before loading.

You don't always have to have `try/except` block. You can remove it if it's not needed.

Reply:
ok sir sure